app/app.go

```diff
@@ -1567,13 +1567,28 @@ func (app *App) Envs() map[string]bindTypes.EnvVar {
 		}
 	}
 	for _, e := range app.ServiceEnvs {
-		mergedEnvs[e.Name] = e.EnvVar
+		envVar := e.EnvVar
+		envVar.ManagedBy = fmt.Sprintf("%s/%s", e.ServiceName, e.InstanceName)
+		mergedEnvs[e.Name] = envVar
 	}
 	sort.Strings(toInterpolateKeys)
 	for _, envName := range toInterpolateKeys {
 		tsuruEnvs.Interpolate(mergedEnvs, toInterpolate, envName, toInterpolate[envName])
 	}
 	mergedEnvs[tsuruEnvs.TsuruServicesEnvVar] = tsuruEnvs.ServiceEnvsFromEnvVars(app.ServiceEnvs)
+
+	mergedEnvs["TSURU_APPNAME"] = bindTypes.EnvVar{
```

Comment:
just gotta make sure those are properly write-protected

Reply:
just checked and they [are](https://github.com/tsuru/tsuru/blob/main/api/app.go#L1175)